kubernetes/samples/scripts/domain-lifecycle/clusterStatus.sh

```diff
@@ -0,0 +1,133 @@
+                                                   -or-
+                                                   $(basename $0) -n myns -d "" -c ""
+   
+  - Dump all clusters and all uid in ns 'myns':    $(basename $0) -n myns
+                                                   -or-
+                                                   $(basename $0) -n myns -d ""
+
+  - Dump all uid across all ns:                    $(basename $0)
+
+Sample output:
+  WKO Cluster Status -n "" -d "" -c "":
+  namespace          domain          cluster    min  max  goal  current  ready
+  ---------          ------          -------    ---  ---  ----  -------  -----
+  sample-domain1-ns  sample-domain1  cluster-1  0    5    2     2        2
+
+EOF
+
+exit $1
+}
+
+# function to dump the domain cluster status in a table
+# $1=ns $2=uid, pass "" to mean "any"
+function clusterStatus() {
+  local __ns="${1:-}"
+  if [ -z "$__ns" ]; then
+    # an empty ns means check all namespaces
+    local __ns_filter="--all-namespaces=true"
+  else
+    local __ns_filter="-n $__ns"
+  fi
+
+  local __uid="${2:-}"
+  local __cluster_name="${3:-}"
+  local __kubernetes_cli="${4:-kubectl}"
+
+  validateKubernetesCliAvailable "$__kubernetes_cli"
+  failIfValidationErrors
+
+  echo
+  echo "WKO Cluster Status -n \"${__ns:-}\" -d \"${__uid:-}\" -c \"${__cluster_name}\":"
+  echo
+
+  (
+    shopt -s nullglob # causes the for loop to silently handle case where no domains match
+
+    echo "namespace domain cluster min max goal current ready"
+    echo "--------- ------ ------- --- --- ---- ------- -----"
+
+    local __val
+    for __val in \
+      $($__kubernetes_cli $__ns_filter get domains \
+        -o=jsonpath='{range .items[*]}{.metadata.namespace}{","}{.metadata.name}{","}{.spec.domainUID}{"\n"}{end}')
+    do
+      local __ns_cur=$(  echo $__val | cut -d ',' -f 1)
+      local __name_cur=$(echo $__val | cut -d ',' -f 2)
+      local __uid_cur=$( echo $__val | cut -d ',' -f 3)
+      local __uid_cur=${uid_cur:-$__name_cur} # uid defaults to name when not set
+      local __jp=''
+
+      [ -n "$__uid" ] && [ ! "$__uid" = "$__uid_cur" ] && continue
+
+      if [ -z "$__cluster_name" ]; then
+        __jp+='{range .status.clusters[*]}'
+      else 
+         __jp+='{range .status.clusters[?(@.clusterName=='\"$__cluster_name\"')]}'
+      fi
+      __jp+='{"'$__ns_cur' "}'
+      __jp+='{"'$__uid_cur' "}'
+      __jp+='{.clusterName}{" NA"}'
+      __jp+='{.minimumReplicas}{" NA"}'
+      __jp+='{.maximumReplicas}{" NA"}'
+      __jp+='{.replicasGoal}{" NA"}'
+      __jp+='{.replicas}{" NA"}'
+      __jp+='{.readyReplicas}{" "}'
+      __jp+='{"\n"}'
+      __jp+='{end}'
+
+      $__kubernetes_cli -n "$__ns_cur" get domain "$__uid_cur" -o=jsonpath="$__jp"
+    done | sed 's/ NA\([0-9][0-9]*\)/ \1/g' | sort --version-sort
+
+  ) | column --table
+ 
+  echo
+}
+
+domainNS=
+domainUID=
+clusterName=
+kubernetesCli=
```

Comment:
Please follow the convention of the other domain lifecycle scripts and default this to $KUBERNETES_CLI env var (plus change the help accordingly).